cfgov/v1/tests/test_templatetags_share.py

```diff
@@ -0,0 +1,113 @@
+import mock
+
+from unittest import TestCase
+from v1.templatetags import share
+from v1.tests.wagtail_pages import helpers
+from v1.models.base import CFGOVPage, CFGOVPagePermissionTester, CFGOVUserPagePermissionsProxy, User
+
+
+class TemplatetagsShareTestCase(TestCase):
+
+    def setUp(self):
+
+        self.page = CFGOVPage(title='a very nice cfgov page')
+        self.shared = False
+        self.live = False
+        self.not_a_page = mock.MagicMock()
+        self.not_a_page.specific = mock.MagicMock()
+
+        helpers.save_new_page(self.page)
+
+        self.user = User(first_name='A', last_name='User')
+        self.user.save()
+
+        self.context = mock.MagicMock()
+        self.request = mock.MagicMock()
+
+        self.context_dict = {}
+
+        def context_getitem(name):
+            return self.context_dict[name]
+
+        def context_setitem(name, value):
+            self.context_dict[name] = value
+
+        self.context.__getitem__.side_effect = context_getitem
+        self.context.__setitem__.side_effect = context_setitem
+
+        self.request.user = self.user
+
+    def tearDown(self):
+
+        self.user.delete()
+
+    def test_is_shared(self):
+
+        result = share.is_shared(self.not_a_page)
+
+        self.assertIsNone(result)
+
+        self.page.shared = True
+        result = share.is_shared(self.page)
+
+        self.assertTrue(result)
+
+        self.page.shared = False
+        result = share.is_shared(self.page)
+
+        self.assertFalse(result)
+
+    def test_get_page_state_url_non_live_non_shared(self):
+
+        self.page.live = False
+        self.page.shared = False
+        self.page.save()
+
+        result = share.get_page_state_url(None, self.page)
+        self.assertIsNone(result)
+
+    def test_get_page_state_url_no_url(self):
+
+        self.page.live = True
+        self.page.url_path = 'somenonvalidpath'
+        self.page.shared = True
+        self.page.save()
+
+        result = share.get_page_state_url(None, self.page)
+        self.assertIsNone(result)
+
+    def test_get_page_state_url_staged(self):
+
+        self.page.live = False
+        self.page.shared = True
+        self.page.save()
+
+        result = share.get_page_state_url(None, self.page)
+        self.assertEqual(result, self.page.url.replace('http://localhost',
+                                                       'http://content.localhost'))
+
+    def test_get_page_state_url_live(self):
```

Comment:
Could you add comments describing each of these tests or otherwise make the test itself read a little bit more descriptively/  For instance, referencing `self.page.url` in the assertion is a little confusing, if that's where our expectation should be, shouldn't we just say 'Live' or whatever there?